packages/components/input/src/use-input.ts

```diff
@@ -250,17 +250,29 @@ export function useInput<T extends HTMLInputElement | HTMLTextAreaElement = HTML
   const hasPlaceholder = !!props.placeholder;
   const hasLabel = !!label;
   const hasHelper = !!description || !!errorMessage;
-  const shouldLabelBeOutside = labelPlacement === "outside" || labelPlacement === "outside-left";
+
+  /*
+    outside -> label is outside only when some placeholder is there
+    outside-top, outside-left-> label is outside regardless of placeholder
+  */
+  const shouldLabelBeOutside =
+    labelPlacement === "outside" ||
+    labelPlacement === "outside-left" ||
+    labelPlacement === "outside-top";
```

Comment:
i got `This comparison appears to be unintentional because the types '"inside" | undefined' and '"outside-top"' have no overlap.ts(2367)`. please double check.